lib/irq.h

```diff
@@ -65,6 +103,16 @@ void metal_irq_unregister(int irq)
 	metal_irq_register(irq, 0, NULL, NULL);
 }
 
+/**
+ * @brief	metal_irq_get_softirq
+ *
+ * Get soft IRQ from triggered from source IRQ
+ *
+ * @param[in] irq source IRQ
+ * @return soft IRQ ID for success, negative value for failure
+ */
+int metal_irq_get_softirq(int irq);
```

Comment:
I don't think irq.[ch] should have any knowledge about soft irq.  Anything related to soft irq should be in softirq.[ch].

Reply:
Removed from irq.[ch], after a double thought, remove it from softirq.[ch] as there is already softIRQ api to allocate soft IRQs. Each driver or IRQ controller should be responsible to maintain  the mapping.